src/.init.lua

```diff
@@ -1,27 +1,246 @@
+        end
+        res = stmt:step()
+        if res == sqlite3.ROW then
+            table.insert(output["leaders"], stmt:get_named_values())
+            if stmt:step() ~= sqlite3.DONE then
+                Log(kLogWarn, "BOARD Internal error (stmt:step after row): %s" % {db:errmsg()})
+                return
+            end
+        elseif res == sqlite3.DONE then
+            table.insert(output["leaders"], false)
+        else
+            Log(kLogWarn, "BOARD Internal error (stmt:step returned %d): %s" % {res, db:errmsg()})
+            return
+        end
+        stmt:reset()
     end
 end
 
+function UpdateBoardImpl(db)
+    local stmt, err, output, board
+    stmt, err = db:prepare([[
+        SELECT nick As name, COUNT(ip) AS count
+        FROM land
+        WHERE ip >= ?1 AND ip <= ?2
+        GROUP BY nick
+        ORDER BY count DESC
+        LIMIT 1
+    ]])
+    if not stmt then
+        Log(kLogWarn, "BOARD prepare board query: %s / %s" % {err or "(null)", db:errmsg()})
+        return
+    end
+    output = {["leaders"]={}}
+    UpdateBoardCrawl(db, output, stmt)
+    stmt:finalize()
+    board = EncodeJson(output)
+    stmt, err = db:prepare([[
+        INSERT INTO cache (key, val) VALUES (?1, ?2)
+        ON CONFLICT (key) DO UPDATE SET (val) = (?2)
+    ]])
+    if not stmt then
+        Log(kLogWarn, "BOARD prepare insert: %s / %s" % {err or "(null)", db:errmsg()})
+        return
+    elseif stmt:bind_values("/board", board) ~= sqlite3.OK then
+        Log(kLogWarn, "BOARD insert bind: %s" % {db:errmsg()})
+    elseif stmt:step() ~= sqlite3.DONE then
+        Log(kLogWarn, "BOARD insert step: %s" % {db:errmsg()})
+    else
+        Log(kLogInfo, "BOARD successfully updated")
+    end
+    stmt:finalize()
+end
+
+function UpdateBoard()
+    local db = ConnectDb()
+    UpdateBoardImpl(db)
+    db:close()
+end
+
+function UpdateBoardWorker()
+    assert(unix.unveil(".", "rwc"))
+    assert(unix.unveil("/var/tmp", "rwc"))
+    assert(unix.unveil("/tmp", "rwc"))
+    assert(unix.unveil(nil, nil))
+    assert(unix.pledge("stdio flock rpath wpath cpath", nil, unix.PLEDGE_PENALTY_RETURN_EPERM))
+    assert(unix.sigaction(unix.SIGINT, OnTerm))
+    assert(unix.sigaction(unix.SIGTERM, OnTerm))
+    while not gotterm do
+        UpdateBoard()
+        unix.nanosleep(30)
+    end
+    Log(kLogInfo, "UpdateBoardWorker() terminating")
+end
+
 function OnServerStart()
     local err
     claims_log, err = unix.open("claims.log", unix.O_WRONLY | unix.O_APPEND | unix.O_CREAT, 0644)
     if err ~= nil then
         Log(kLogFatal, string.format("error opening claim log: %s", err))
     end
+    local pid = assert(unix.fork())
+    if pid == 0 then
+        UpdateBoardWorker()
+        unix.exit(0)
+    end
+end
+
+function OnWorkerStart()
+    db = sqlite3.open("db.sqlite3")
+    db:busy_timeout(1000)
+    db:exec[[PRAGMA journal_mode=WAL]]
+    db:exec[[PRAGMA synchronous=NORMAL]]
+
+    -- TODO(jart): Must we do this?
```

Comment:
I see that a warmup query is shown as recommended guidance on the redbean docs so I assume you've determined the answer is yes?